File: app/src/main/java/ru/mamykin/foreignbooksreader/di/modules/AppModule.java
package ru.mamykin.foreignbooksreader.di.modules;

import android.content.Context;
import android.support.annotation.NonNull;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Creation date: 5/29/2017
 * Creation time: 11:39 AM
 * @author Andrey Mamykin(mamykin_av)
 */
@Module
public class AppModule {
    private Context context;

    public AppModule(@NonNull Context context) {
        this.context = context;
    }

    @Provides
    @Singleton
    Context provideContext() {
        return context;
    }
}

File: app/src/main/java/ru/mamykin/foboreader/core/di/modules/AppModule.kt
package ru.mamykin.foboreader.core.di.modules

import android.content.Context
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
@Singleton
class AppModule(private val context: Context) {

    @Provides
    @Singleton
    fun provideContext(): Context = context
}